core/src/main/java/hudson/slaves/ComputerRetentionWork.java

```diff
@@ -49,9 +51,29 @@ public class ComputerRetentionWork extends PeriodicWork {
      */
     private final Map<Computer, Long> nextCheck = new WeakHashMap<>();
 
+    private ScheduledFuture<?> pending;
+
+    @Override
+    protected void schedulePeriodicWork() {
+        pending = Timer.get().scheduleAtFixedRate(this, this.getInitialDelay(), this.getRecurrencePeriod(), TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Restart the periodic check.
+     * The previous run may still be pending. We do not need to wait for its completion as all calls of
+     * {@link RetentionStrategy#check} inside the runner are performed in a queue with a lock.
+     * Per the docs on {@link RetentionStrategy#check}, it is OK to recheck earlier or later than requested.
+     */
```

Comment:
```suggestion
     * @since TODO
     */
```

If this is intended to be not restricted and therefore public API, please add a since annotation.

Reply:
I've added the since annotation via 0fefa24987f2ba752eb793a37bf554c0ea03e5d8.